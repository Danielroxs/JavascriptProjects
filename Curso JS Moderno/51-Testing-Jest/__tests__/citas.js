import Citas from "../js/classes/Citas"

describe('Probar la clase de citas', () => {

    const citas = new Citas()
    const id = 1

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Juan',
            telefono: '19030913',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'Solo duerme'
        }


        citas.agregarCita(citaObj)

        // Prueba
        expect(citas).toMatchSnapshot()

    })

    test('Actualizar cita', () => {

        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Juan',
            telefono: '19030913',
            fecha: '10-12-2020',
            hora: '10:30',
            sintomas: 'Solo duerme'
        }

        citas.editarCita(citaActualizada)

        expect(citas).toMatchSnapshot()
    })

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot()
    })

})