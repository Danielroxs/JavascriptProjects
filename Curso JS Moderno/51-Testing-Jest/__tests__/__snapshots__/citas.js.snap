// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1,
      "mascota": "Nuevo nombre",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "19030913",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo duerme",
      "telefono": "19030913",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
